rerun/src/main/java/org/apache/falcon/rerun/handler/RetryConsumer.java

```diff
@@ -43,14 +43,18 @@ public RetryConsumer(T handler) {
     protected void handleRerun(String clusterName, String jobStatus,
                                RetryEvent message, String entityType, String entityName) {
         try {
-            if (!jobStatus.equals("KILLED")) {
-                LOG.debug("Re-enqueing message in RetryHandler for workflow with same delay as job status is running:"
-                        + " {}", message.getWfId());
+            // Can happen when user does a manual re-run in-between retries.
+            if (jobStatus.equals("RUNNING") || jobStatus.equals("PREP")) {
+                LOG.debug("Re-enqueing message in RetryHandler for workflow with same delay as "
+                        + "job status is {} for : {}", jobStatus, message.getWfId());
                 message.setMsgInsertTime(System.currentTimeMillis());
                 handler.offerToQueue(message);
                 return;
+            } else if (jobStatus.equals("SUSPENDED") || jobStatus.equals("SUCCEEDED")) {
+                LOG.debug("Not retrying workflow {} anymore as it is in {} state. ", message.getWfId(), jobStatus);
```

Comment:
I thought about it too. My thought process was, if it has been suspended, then, when the user resumes it, it will either fail or succeed. If it fails, the retry will anyway kick in (as a new RetryEvent). Since we derive the retry count from the runId, we won't exceed the attempts. So, I thought there is no need to constantly monitor a SUSPENDED instance for retries.

Am I missing something?